examples/staking-broker/src/wasm.rs

```diff
@@ -0,0 +1,266 @@
+// This file is part of Gear.
+//
+// Copyright (C) 2024 Gear Technologies Inc.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+//
+// This is a very basic implementation of a staking broker program for demo purpose only.
+// It accepts users messages and bond, unbond or nominate some validators on their behalf.
+// It is not concerned with some important constraints like unbonding period etc., so a real
+// implementation of a liquid staking contract, for example, would be more complex.
+
+use gbuiltin_staking::*;
+use gstd::{debug, msg, prelude::*, ActorId};
+use hashbrown::HashMap;
+use hex_literal::hex;
+use parity_scale_codec::Encode;
+
+// Staking proxy builtin actor program id (hardcoded for all runtimes)
+const BUILTIN_ADDRESS: ActorId = ActorId::new(hex!(
+    "77f65ef190e11bfecb8fc8970fd3749e94bed66a23ec2f7a3623e785d0816761"
+));
```

Comment:
> This is different from
> 
> https://github.com/gear-tech/gear/blob/4638ed4f9cc9d7fab848bebc5d6e7465d5275ddd/gbuiltins/staking/src/lib.rs#L40-L42
> 

Right, the current `BuiltinId` is generated using a different seed comparing to the old way (now we don't use any `PalletId` whatsoever, hence the difference). To start with, I'll align the comment with the actual ID.

Not quite sure about hardcoding the address in `gbuiltin` crate: sure all builtin ids has to be published somewhere (#3854), but inherently they are not hardcoded but rather generated in the runtime. Most likely it is not going to be the case but it is still possible that the ID generation algo changes, which will make the `gbulitin` go out of sync with the runtime if it has this constant hardcoded.

Reply:
> Most likely it is not going to be the case but it is still possible that the ID generation algo changes

Yeah, for this, I was thinking about calculating it in `build.rs` and export, etc, seems dirty in development but if we have some kind of this implementations ( `gbuiltin::STAKING` ), it would be super neat for developers, they can import the addresses it directly from our library instead of copying it from somewhere and updating it manually

btw if the addresses of our builtin programs are `versioned`, if it is not versioned and we generated different addresses in different builds, it could cause a lot of programs in the development of gear programs for users